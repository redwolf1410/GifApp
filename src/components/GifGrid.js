import React,{useEffect,useState} from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifs } from "../helpers/getGifs";
import { useFechGifs } from "../hooks/useFechGifs";
import PropTypes from "prop-types";
export const GifGrid = ({ category }) => {

    const {data:images, loading } = useFechGifs(category);

    return (
        <>
        <h3>{category}</h3>

        { loading ? 'Loading...' : 
         <div className='card-grid'>
            
            {
                images.map( img => (
                        <GifGridItem 
                        key={img.id}
                        {...img}/>     
                        )
                )
            }
        </div> }
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}