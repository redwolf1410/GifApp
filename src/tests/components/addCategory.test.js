import {AddCategory} from '../../components/AddCategory';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import React  from 'react';



describe('tests for AddCategory', () => { 
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories = {setCategories} />)
    
    beforeEach( ()=> {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories = {setCategories} />)

    }
    )
    test('should be the same as the snapshot', () => { 

        expect(wrapper).toMatchSnapshot();

     })

     test('should change textbox', () => { 
         const input = wrapper.find('input');
         const value = 'Hola mundo';

         input.simulate('change',{ target: {value}});

         expect(wrapper.find('p').text().trim()).toBe(value)
      })
      test('should not post info with submit', () => { 
          wrapper.find('form').simulate('submit',{preventDefault(){}}) 
          expect(setCategories).not.toHaveBeenCalled(); 

       })
       test('should call setcategories abd clean textbox', () => { 
           const value = 'Hola mundo'
           wrapper.find('input').simulate('change',{target:{value} })

           wrapper.find('form').simulate('submit',{preventDefault(){} })
           expect(setCategories).toHaveBeenCalled()
           expect(wrapper.find('input').prop('value')).toBe('');

        })

 })