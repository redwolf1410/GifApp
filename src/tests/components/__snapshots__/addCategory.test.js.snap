// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for AddCategory should be the same as the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
